### ISSUE TITLE
Regression SB16 Win31 driver gives MIDIPORT issue with cpu_vm+cpu_vm_dpmi emulated

### ISSUE THREAD

**Issue by jschwartzenberg** [dosemu2/dosemu2#1157]
**Describe the bug**
With:
```
$_cpu_vm = "emulated"
$_cpu_vm_dpmi = "emulated"
```
Plus either `$_cpu_emu = "full"` or `$_cpu_emu = "fullsim"`, installing th SB16 driver gives the following screen on win31 startup:
![afbeelding](https://user-images.githubusercontent.com/877112/82130400-53e6a080-97cb-11ea-8594-16926d1b6d02.png)

**To Reproduce**
1. Install win31
2. Unpack https://github.com/jschwartzenberg/dos-download-scripts/raw/master/sb16w31/sb16.tar.xz to `c:\`
3. Run `win c:\sb16\winsetup.exe`

**Attach program/game binaries or provide an URL**
https://github.com/jschwartzenberg/dos-download-scripts/raw/master/sb16w31/sb16.tar.xz

**Attach the log**
[boot.log](https://github.com/dosemu2/dosemu2/files/4639235/boot.log)

**Additional info**
This doesn't occur with KVM.

**Comment by stsp** [comment]
I suspect this doesn't occur with vm86sim/vm86
cpuemu modes either?

**Comment by stsp** [comment]
Certainly not a regression, $_cpu_vm_dpmi = "emulated"
didn't work ever before.

**Comment by jschwartzenberg** [comment]
With these I get the same:
```
$_cpu_vm = "emulated"
$_cpu_vm_dpmi = "emulated"
$_cpu_emu = "vm86sim"
```

But you're right, I completely forgot dpmi was typically not emulated. Which variants make sense to test extra?

**Comment by stsp** [comment]
$_cpu_vm_dpmi = "native"

**Comment by jschwartzenberg** [comment]
Yeah, that works. Indeed. So it's up to you if this specific report makes sense. It seems that things do mostly work though. (With dpmi to kvm it win31 crashes dosemu at some point btw.)

**Comment by stsp** [comment]
For example on exit.
Which is why kvm dpmi is not yet
a default, it is not very stable. But
could you please check it is not a
regression, eg please try 2757fed

**Comment by jschwartzenberg** [comment]
For that commit `$_cpu_vm_dpmi = "emulated"` doesn't even start DOS, but `$_cpu_vm_dpmi = "kvm"` works fine, no crashes.

**Comment by stsp** [comment]
So KVM regressed.
Could you please bisect?

**Comment by jschwartzenberg** [comment]
Tried, but can't reproduce anymore... seems it's a tricky one. I'll test intensively once you give me a go for DPMI KVM ;)

**Comment by stsp** [comment]
What do you mean?

**Comment by jschwartzenberg** [comment]
I means that when you say that DPMI KVM should be stable, I will start testing it more.

**Comment by stsp** [comment]
But its about bisect, it worked in the past.
How can it became stable w/o bisecting?

**Comment by jschwartzenberg** [comment]
I tried to reproduce it for the bisect, but failed. I can't reproduce the crash at all anymore, also with the latest version it works now.

**Comment by stsp** [comment]
Yes, I can understand this. :)
What I can't get is how can I help
by give you "a go for DPMI KVM".
I'd like to, but you won't be able to
reproduce nevertheless.

**Comment by jschwartzenberg** [comment]
I mean more in general. If you tell me it needs testing, I'll just test with it as well :)

**Comment by stsp** [comment]
So this was mentioned to be
no longer reproducible.
